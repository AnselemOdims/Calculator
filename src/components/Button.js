import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { selector, sym } = this.props;
    return (
      <button type="button" className={selector.join(' ')}>{sym}</button>
    );
  }
}

Button.propTypes = {
  selector: PropTypes.arrayOf(PropTypes.string).isRequired,
  sym: PropTypes.string.isRequired,
};

export default Button;
