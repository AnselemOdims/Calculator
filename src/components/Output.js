import PropTypes from 'prop-types';

const Output = ({ total, operation, next }) => (
  <div className="output">
    {total}
    {operation}
    {next}
  </div>
);

Output.defaultProps = {
  operation: null,
  next: null,
};

Output.propTypes = {
  total: PropTypes.string.isRequired,
  operation: PropTypes.string,
  next: PropTypes.string,
};

export default Output;
