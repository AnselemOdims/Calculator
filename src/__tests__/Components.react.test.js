/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';
import Output from '../components/Output';

describe('Button', () => {
	const sym = '+';
	const selector = ['digits', 'orange'];
	const handleClick = () => {};
	const button = renderer.create(
		<Button sym={sym} handleBtnClick={handleClick} selector={selector} />
	);
	let tree = button.toJSON();

	test('Button mounts when called', () => {
		expect(tree).toMatchSnapshot();
	});

	test('onClick function works properly', () => {
		tree.props.onClick();
		tree = button.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('Output', () => {
	const total = '0';
	const operation = '+';
	const next = '0';
	test('Output display appropriately', () => {
		const output = renderer.create(
			<Output total={total} operation={operation} next={next} />
		);
    const tree = output.toJSON();
    expect(tree).toMatchSnapshot();
	});
});
