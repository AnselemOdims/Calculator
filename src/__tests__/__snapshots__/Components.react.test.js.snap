// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button mounts when called 1`] = `
<button
  className="digits orange"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;

exports[`Button onClick function works properly 1`] = `
<button
  className="digits orange"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;

exports[`Output Output display appropriately 1`] = `
<div
  className="output"
>
  0
  +
  0
</div>
`;
